app/src/test/java/org/simple/clinic/teleconsultlog/teleconsultrecord/screen/TeleconsultRecordEffectHandlerTest.kt

```diff
@@ -78,4 +91,45 @@ class TeleconsultRecordEffectHandlerTest {
 
     verifyZeroInteractions(uiActions)
   }
+
+  @Test
+  fun `when create teleconsult record effect is received, then create the teleconsult record`() {
+    // given
+    val patientUuid = UUID.fromString("cd8c77c4-67de-482a-bfd0-2b035291b45d")
+    val teleconsultRecord = TestData.teleconsultRecord(
+        id = uuidGenerator.v4(),
+        patientId = patientUuid,
+        medicalOfficerId = user.uuid,
+        teleconsultRecordInfo = TestData.teleconsultRecordInfo(
+            recordedAt = Instant.parse("2018-01-01T00:00:00Z"),
+            teleconsultationType = Audio,
+            patientTookMedicines = Yes,
+            patientConsented = Yes,
+            medicalOfficerNumber = null
+        ),
+        teleconsultRequestInfo = null,
+        createdAt = Instant.now(utcClock),
+        updatedAt = Instant.now(utcClock)
+    )
+
+    doNothing().whenever(teleconsultRecordRepository).createTeleconsultRecordForMedicalOfficer(
+        teleconsultRecordId = teleconsultRecord.id,
+        patientUuid = teleconsultRecord.patientId,
+        medicalOfficerId = teleconsultRecord.medicalOfficerId,
+        teleconsultRecordInfo = teleconsultRecord.teleconsultRecordInfo!!
+    )
+
+    // when
+    effectHandlerTestCase.dispatch(CreateTeleconsultRecord(
+        patientUuid = patientUuid,
+        teleconsultationType = Audio,
+        patientTookMedicine = Yes,
+        patientConsented = Yes
+    ))
+
+    // then
+    effectHandlerTestCase.assertOutgoingEvents(TeleconsultRecordCreated)
```

Comment:
We should also verify the side effect of creating the record on the repository mock, correct?